### PR TITLE
Add actions.py and parser.py to project

### PR DESCRIPTION
**Pull request by Hhamid1991**
Update `parser.py` file.

### CODE REVIEW

File: sunrise/actions.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+
+
+class Action :
+	pass
+
+
+class Calculator(Action) :
+	pass
+
```
Comment (pylover) [inline sunrise/actions.py:3]: PEP8

**Review by pylover — CHANGES_REQUESTED** [review]

**Comment by Hhamid1991** [comment]
Add `ParserError.py` to project and modify `parser.py` and test `parser.py`

File: sunrise/parser.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import re
+from sunrise.actions import Calculator
+
+
+patternslist = [
+    ('[-+]?[0-9]+\.?[0-9]*\s*[-+*\/]\s*[-+]?[0-9]+\.?[0-9]*', Calculator)
```
Comment (pylover) [inline sunrise/parser.py]: Use `r'....'` prefix for regex pattern.

Comment (pylover) [inline sunrise/parser.py]: Use `re.compile(...)`.

File: sunrise/parser.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+]
+
```
Comment (pylover) [inline sunrise/parser.py:12]: Use double empty lines.

File: sunrise/parser.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+def parse(inputstring):
+    for pattern, action in patternslist:
+        if re.search(pattern, inputstring):
+            return action
+    raise ParserException(inputstring)
```
Comment (pylover) [inline sunrise/parser.py]: Add an extra blank line above.

File: sunrise/parser.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+from sunrise.actions import Calculator
+
+
+patternslist = [
+    ('[-+]?[0-9]+\.?[0-9]*\s*[-+*\/]\s*[-+]?[0-9]+\.?[0-9]*', Calculator)
+]
+
+def parse(inputstring):
+    for pattern, action in patternslist:
+        if re.search(pattern, inputstring):
```
Comment (pylover) [inline sunrise/parser.py]: `pattern.search(inputstring)`

Comment (pylover) [inline sunrise/parser.py]: Use `match` instead  of `search`.

File: sunrise/actions.py [hunk 6]
```diff
@@ -0,0 +1,6 @@
+class Action :
```
Comment (pylover) [inline sunrise/actions.py]: Unallowed space before `:`.

File: sunrise/actions.py [hunk 7]
```diff
@@ -0,0 +1,6 @@
...
+	pass
+
+
+class Calculator(Action) :
```
Comment (pylover) [inline sunrise/actions.py]: Same as the above.

File: sunrise/parser.py [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import re
+from sunrise.actions import Calculator
```
Comment (pylover) [inline sunrise/parser.py]: PEP8, input order.

**Review by pylover — CHANGES_REQUESTED** [review]

File: sunrise/parseExceptions.py [hunk 9]
```diff
@@ -0,0 +1,3 @@
+class ParseError(Exception):
+    def __init__(self, s):
```
Comment (pylover) [inline sunrise/parseExceptions.py]: `s`?

**Review by pylover — APPROVED** [review]

**Pull request merged**